util/conditions/patch_test.go

```diff
@@ -112,6 +143,30 @@ func TestApply(t *testing.T) {
 		want    clusterv1.Conditions
 		wantErr bool
 	}{
+		{
+			name:    "success on Apply with nil patch",
```

Comment:
Q: is the name of this test correct?

Q2: do we need this test? Given that this is not a supported use case (as NewPatch returns an err). If we want to keep the test please add a comment that this case is not supported. Just not that folks get the impression from the unit test that we support this case.

Reply:
Just stepped through that test case. I would really prefer if we drop this test case. I don't think that this is a behavior we should test / enforce through a test case